book/src/proptest-state-machine/getting-started.md

```diff
@@ -0,0 +1,176 @@
+
+> Given that there are thought to be in the region of another four million species that we have not yet even named, there is no doubt that scientists will be kept happily occupied studying them for millennia, so long as the insects remain to be studied. Would the world not be less rich, less surprising, less wonderful, if these peculiar creatures did not exist?
+>
+> -- <cite>Dave Goulson, Silent Earth</cite>
+
+So let's leave bugs alone and only squash defects instead!
+
+Because the output of a failed test case can be a bit hard to read, it is often convenient to print the transitions. You can do that by simply setting the `proptest_config`'s `verbose` to `1` or higher. Again, if you don't want to keep this in your test's config or if you'd prefer to override the config, you could also use the `PROPTEST_VERBOSE` environment variable instead.
+
+Another helpful config option that is good to know about is `timeout` (`PROPTEST_TIMEOUT` via an env var) for tests that may take longer to execute.
+
+## How does it work
+
+This section goes into the inner workings of how the state machine is implemented, omitting some less interesting details. If you're only interested in using it, you can consider this section an optional read.
+
+The `ReferenceStateMachine::sequential_strategy` sets up a `Sequential` strategy that generates a sequence of transitions from the definition of the `ReferenceStateMachine`. The acceptability of each transition in the sequence depends on the current state of the state machine and `ReferenceStateMachine::preconditions`, if any. The state is updated by the transitions with the `ReferenceStateMachine::apply` function.
+
+The `Sequential` strategy is then fed into Proptest like any other strategy via the `prop_state_machine!` macro and it produces a `Vec<Transition>` that gets passed into `StateMachineTest::test_sequential` where it is applied one by one to the SUT. Its post-conditions and invariants are checked during this process and if a failing case is found, the shrinking process kicks in until it can shrink no longer.
+
+The shrinking strategy which is defined by the associated `type Tree = SequentialValueTree` of the `Sequential` strategy is to iteratively apply `Shrink::InitialState`, `Shrink::DeleteTransition` and `Shrink::Transition` (this can be found in `proptest/src/strategy/state_machine.rs`):
+
+1. We start by trying to delete transitions from the back of the list until we can do so no further (the list has reached the `min_size` - that is the variable that gets set from the chosen range for the number of transitions in the `prop_state_machine!` invocation).
+2. Then, we again iteratively attempt to shrink the individual transitions, but this time starting from the front of the list from the first transition to be applied.
+3. Finally, we try to shrink the initial state until it's not possible to shrink it any further.
```

Comment:
I think this section will need to be updated to reflect this change https://github.com/proptest-rs/proptest/blob/ca308b0362336d7bf0e387052daadf0251adcecb/proptest-state-machine/src/strategy.rs#L124-L137